smack-core/src/main/java/org/jivesoftware/smack/altConnections/HttpLookupMethod.java

```diff
@@ -0,0 +1,187 @@
+                String namespace = parser.getNamespace();
+                String rel = parser.getAttributeValue("rel");
+
+                if (!namespace.equals(xrd_namespace) || !name.equals("Link") || !rel.equals(relation)) {
+                    tag = parser.nextTag();
+                    continue loop;
+                }
+                String endpointUri = parser.getAttributeValue("href");
+                uriList.add(new URI(endpointUri));
+                tag = parser.nextTag();
+                break;
+            case END_ELEMENT:
+                if (parser.getDepth() == initialDepth) {
+                    break loop;
+                }
+                tag = parser.nextTag();
+                break;
+            }
+        }
+        return uriList;
+    }
+
+    /**
+     * Get remote endpoints for the provided LinkRelation from provided XmlPullParser.
+     *
+     * @param parser XmlPullParser that contains LinkRelations
+     * @param relation type of endpoints specified by the given LinkRelation
+     * @return list of endpoints
+     * @throws IOException exception due to input/output operations
+     * @throws XmlPullParserException exception encountered during XML parsing
+     * @throws URISyntaxException exception to indicate that a string could not be parsed as a URI reference
+     */
+    public static List<URI> parseXrdLinkReferencesFor(XmlPullParser parser, LinkRelation relation) throws IOException, XmlPullParserException, URISyntaxException {
+        return parseXrdLinkReferencesFor(parser, relation.attribute);
+    }
+
+    /**
+     * Get remote endpoints for the given LinkRelation from host.
+     *
+     * @param xmppServiceAddress address of host
+     * @param relation LinkRelation as a string specifying type of connection
+     * @return list of endpoints
+     * @throws IOException exception due to input/output operations
+     * @throws XmlPullParserException exception encountered during XML parsing
+     * @throws URISyntaxException exception to indicate that a string could not be parsed as a URI reference
+     */
+    public static List<URI> lookup(DomainBareJid xmppServiceAddress, String relation) throws IOException, XmlPullParserException, URISyntaxException {
+        final int buffer_size = 256;
+        InputStream inputStream = getXrdStream(xmppServiceAddress);
+        int remainingBytes = inputStream.available();
+        ByteArrayOutputStream obtainedXml = new ByteArrayOutputStream();
+
+        // Keep reading from the input stream until read completely
+        while (remainingBytes > 0) {
+            byte[] byteBuffer = new byte[buffer_size];
+            int bytes_read = inputStream.read(byteBuffer);
+            obtainedXml.write(byteBuffer, 0, bytes_read);
+            remainingBytes = remainingBytes - bytes_read;
+        }
+        inputStream.close();
+        XmlPullParser parser = PacketParserUtils.getParserFor(obtainedXml.toString());
```

Comment:
Again, there is no need to additionally buffer the input stream, simply use `PacketParserUtils.getParserFor(InputStream)`.